### ISSUE TITLE
Misleading comment in configuration.nix

### ISSUE THREAD

**Issue by earksiinni** [NixOS/nixpkgs#88948]
**Describe the bug**
The NixOS `system.stateVersion` option in configuration.nix has the following comment:

> This value determines the NixOS release with which your system is to be
> compatible, in order to avoid breaking some software such as database
> servers. You should change this only after NixOS release notes say you
> should.

However, [the release notes](https://nixos.org/nixos/manual/release-notes.html) don't say anything about changing the `stateVersion`. I just upgraded my system from 19.03 to 20.03 and I'm not sure whether I'm supposed to update this option.

**To Reproduce**
N/A

**Expected behavior**
A comment that accurately describes under what conditions `system.stateVersion` should be changed and to what version it should be changed, or an updated release notes that explicitly instructs the user when and how to change `system.stateVersion`.

**Screenshots**
N/A

**Additional context**

**Notify maintainers**

(Not sure who's responsible for this file and/or the release notes.)

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.42, NixOS, 20.03.1950.48723f48ab9 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
 - channels(root): `"nixos-20.03.1950.48723f48ab9"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by OPNA2608** [comment]
Mentioning relevant PR #74138.

tl;dr You should never need to change the version, *unless* the release notes one day tell you to do so, which they normally won't.

---

> stateVersion is used by nixos for state whose default changed throughout different versions, e.g. a database directory now being in /var/db instead of /var/lib/db.
> So there is a switch: dbDir = if stateVersion > 18.03 then /var/db else /var/lib/db.
> 
> Updating this variable would mean that suddenly it thinks the db is in a different place, so it breaks.
> You can update it if you really want it, but then you need to go through the list of changes and manually migrate everything.
> 
> That means moving the database from /var/db to /var/lib/db yourself.
> Just don’t update it, it’s not worth it.
> 
> The only thing it gets you is that the string “17.09” isn’t in your config anymore and potential problems, literally nothing else.

And stateVersion's description:
https://github.com/NixOS/nixpkgs/blob/147aded7df024288a387e1dde290f79f964e41bf/nixos/modules/misc/version.nix#L56-L73

**Comment by earksiinni** [comment]
@OPNA2608 thank you for the clarification. The description that you linked to is _much_ more clear than the comment in configuration.nix, which leads the reader to believe that this setting is meant to be changed regularly (as noted in https://github.com/NixOS/nixpkgs/pull/74138, which you linked to).

I think that this is a documentation bug. (IMHO, any time I see something like "`Did you read the comment?`", which is also in configuration.nix, that's a bad doc smell.)

Closing this in favor of https://github.com/NixOS/nixpkgs/pull/74138.

**Comment by maralorn** [comment]
I it is worth noting that the comment in the configuration.nix has been changed in january 2020:

https://github.com/NixOS/nixpkgs/pull/77279

**Comment by maralorn** [comment]
Of course people will have the older comment from older installs in their systems for quite a while.
